File: src/main/java/com/vaadin/demo/component/rangeslider/RangeSliderMinMaxVisible.java
package com.vaadin.demo.component.rangeslider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.RangeSlider;
import com.vaadin.flow.component.slider.RangeSliderValue;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("range-slider-min-max-visible")
public class RangeSliderMinMaxVisible extends Div {

    public RangeSliderMinMaxVisible() {
        // tag::snippet[]
        RangeSlider rangeSlider = new RangeSlider("Temperature",
                0, 100, new RangeSliderValue(20, 80));
        // rangeSlider.setMinMaxVisible(true);
        add(rangeSlider);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<RangeSliderMinMaxVisible> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/range-slider/range-slider-min-max-visible.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider/vaadin-range-slider.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('range-slider-min-max-visible')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-range-slider
        label="Temperature"
        min="0"
        max="100"
        .value="${[20, 80]}"
        min-max-visible
      ></vaadin-range-slider>
      <!-- end::snippet[] -->
    `;
  }
}
